### PR TITLE
Implemented Signed Pow Hint

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
Fixes: #441

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_math_test.go [hunk 1]
```diff
@@ -545,6 +545,36 @@ func TestZeroHintMath(t *testing.T) {
...
+		"SignedPow": {
+			// Zero value - assertion failed, any other - nothing.
+			{
+				operanders: []*hintOperander{
+					{Name: "base", Kind: apRelative, Value: feltInt64(0)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSignedPowHint(ctx.operanders["base"])
+				},
+				errCheck: errorTextContains("Cannot raise 0 to a negative power"),
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_math_test.go:557]: can you put the entire error text here?

**Comment by MaksymMalicki** [comment]
Some nitpicks, but overall, looks good!

File: pkg/hintrunner/zero/zerohint_math_test.go (hunk 1 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_math_test.go:557]: the full text is this : "assertion `Cannot raise 0 to a negative power` failed". I think the current impl is just fine. Though I can change if think that would be better. I was just following the design of `newSplitIntHint`.

File: pkg/hintrunner/zero/zerohint_math_test.go [hunk 2]
```diff
@@ -545,6 +545,36 @@ func TestZeroHintMath(t *testing.T) {
...
+			},
+			{
```
[the remark this answers is not in the document]
Comment (Sh0g0-1758) [reply pkg/hintrunner/zero/zerohint_math_test.go:559]: Well yeah the comment on line 549 : 
` // Zero value - assertion failed, any other - nothing. `
does just that. Its meant to cover all cases.

**Comment by Sh0g0-1758** [comment]
@TAdev0 , @MaksymMalicki , is there anything else that needs to be rectified ? Else, it would be great if you could approve and I will merge then.

**Review by MaksymMalicki — APPROVED** [review]

File: pkg/hintrunner/zero/hintcode.go [hunk 3]
```diff
@@ -40,6 +40,8 @@ const (
 
 	unsignedDivRemCode string = "from starkware.cairo.common.math_utils import assert_integer\nassert_integer(ids.div)\nassert 0 < ids.div <= PRIME // range_check_builtin.bound, \\\n    f'div={hex(ids.div)} is out of the valid range.'\nids.q, ids.r = divmod(ids.value, ids.div)"
 
+	signedPowCode string = "assert ids.base != 0, 'Cannot raise 0 to a negative power.'"
```
Comment (TAdev0) [inline pkg/hintrunner/zero/hintcode.go:43]: Why not adding it after PowCode in pow() hints ?

Comment (TAdev0) [reply pkg/hintrunner/zero/hintcode.go:43]: Anyway we will clean the file later

**Review by TAdev0 — APPROVED** [review]

**Pull request merged**